Source/DafnyCore/Verifier/BoogieGenerator.ExpressionTranslator.cs

```diff
@@ -1485,6 +1485,32 @@ public ExpressionTranslator(ExpressionTranslator etran, Boogie.Expr heap)
               UnboxingCastExpr e = castExpr;
               return BoogieGenerator.CondApplyUnbox(GetToken(e), TrExpr(e.E), e.FromType, e.ToType);
             }
+          case DecreasesToExpr decreasesToExpr:
+            var oldArray = decreasesToExpr.OldExpressions.ToArray();
```

Comment:
Once these fields are no longer `IEnumerable`, you won't need the `ToArray` here.